server/v2/stf/core_event_service.go

```diff
@@ -39,12 +37,20 @@ type eventManager struct {
 // Emit emits an typed event that is defined in the protobuf file.
 // In the future these events will be added to consensus.
 func (em *eventManager) Emit(tev transaction.Msg) error {
-	res, err := TypedEventToEvent(tev)
-	if err != nil {
-		return err
+	event := event.Event{
+		Type: gogoproto.MessageName(tev),
+		Data: func() (json.RawMessage, error) {
+			buf := new(bytes.Buffer)
+			jm := &jsonpb.Marshaler{OrigName: true, EmitDefaults: true, AnyResolver: nil}
+			if err := jm.Marshal(buf, tev); err != nil {
+				return nil, err
+			}
+
+			return buf.Bytes(), nil
+		},
```

Comment:
So there is a function Event.Attributes which should also be filled in. This function I believe should lazily call `TypedEventToEvent`

Reply:
I see, I was doing this logic in the CometBFT server, but even better if it canbe done here directly 👍🏾